aptos-move/e2e-move-tests/src/tests/aggregator_v2.data/pack/sources/aggregator_v2_test.move

```diff
@@ -184,76 +184,94 @@ module 0x1::aggregator_v2_test {
+    public entry fun materialize<Element: store + drop>(addr: address, use_type: u32, i: u64) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
         call_read<Element>(addr, use_type, i);
     }
 
     /// Checks that the ith aggregator has expected value. Useful to inject into
     /// transaction block to verify successful and correct execution.
-    public entry fun check<Element: store + drop>(_account: &signer, addr: address, use_type: u32, i: u64, expected: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
+    public entry fun check<Element: store + drop>(addr: address, use_type: u32, i: u64, expected: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
         let actual = call_read<Element>(addr, use_type, i);
         assert!(actual == expected, ENOT_EQUAL)
     }
 
-    public entry fun new_add<Element: drop + copy + store>(account: &signer, addr: address, use_type: u32, i: u64, limit: Element, a: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
-        new(account, addr, use_type, i, limit);
-        add(account, addr, use_type, i, a);
+    public entry fun new_add<Element: drop + copy + store>(addr: address, use_type: u32, i: u64, limit: Element, a: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
+        new(addr, use_type, i, limit);
+        add(addr, use_type, i, a);
     }
 
-    public entry fun sub_add<Element: store + drop>(_account: &signer, addr: address, use_type: u32, i: u64, a: Element, b: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
+    public entry fun sub_add<Element: store + drop>(addr: address, use_type: u32, i: u64, a: Element, b: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
         for_element_mut<Aggregator<Element>, bool>(addr, use_type, i, |aggregator| {
             aggregator_v2::sub(aggregator, a);
             aggregator_v2::add(aggregator, b);
             true
         });
     }
 
-    public entry fun add_sub<Element: store + drop>(_account: &signer, addr: address, use_type: u32, i: u64, a: Element, b: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
+    public entry fun add_if_at_least<Element: store + drop>(addr: address, use_type: u32, i: u64, a: Element, b: Element) acquires AggregatorInResource, AggregatorInTable, AggregatorInResourceGroup {
+        let is_at_least = for_element_ref<Aggregator<Element>, bool>(addr, use_type, i, |aggregator| {
+            aggregator_v2::is_at_least(aggregator, a)
+        });
+
+        if (is_at_least) {
+            add<Element>(addr, use_type, i, b);
+        }
+
+        // issue with type inference of lambda or downcasting from mut to non-mut?
```

Comment:
what's this?

Reply:
lambda doesn't seem to work when first instruction infers &T, and second infers &mut T, it doesn't compile